src/App.tsx

```diff
@@ -15,36 +15,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+export enum SortKeys {
+  ALPHABETICAL = 'Alphabetical',
+  LENGTH = 'Length',
+}
+
+export const App: FC = () => {
+  const [sort, setSort] = useState<string>('');
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const getSortedList = (): string[] => {
+    let sortedGoods = [...goodsFromServer];
+
+    switch (sort) {
+      case SortKeys.ALPHABETICAL:
+        sortedGoods.sort();
+        break;
+      case SortKeys.LENGTH:
+        sortedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedGoods = [...goodsFromServer];
+    }
+
+    return isReversed ? sortedGoods.reverse() : sortedGoods;
+  };
+
+  const toggleReverse = (): void => {
+    setIsReversed(prev => !prev);
+  };
+
+  const isOriginalOrder = sort === '' && !isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sort === SortKeys.ALPHABETICAL ? '' : 'is-light'}`}
+          onClick={(): void => setSort(SortKeys.ALPHABETICAL)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sort === SortKeys.LENGTH ? '' : 'is-light'}`}
+          onClick={(): void => setSort(SortKeys.LENGTH)}
```

Comment:
```suggestion
          onClick={() => setSort(SortKeys.LENGTH)}
```
You don't need to specify a type every time, in this case typescript understands